docs/howto/weather-and-distance-tool/README.md

```diff
@@ -0,0 +1,411 @@
+    Anchor,
+    PluginV2,
+)
+from ayx_python_sdk.providers.amp_provider.amp_provider_v2 import AMPProviderV2
+
+import pyarrow as pa
+import requests
+import jsonpath_rw_ext as jp_ext
+```
+
+> :information_source: Note that `pyarrow as pa` is imported in the default plugin only if TYPE_CHECKING is true. This should be changed to import always.
+
+### Initialization
+Now, it's time to update the plugin's `__init__` method and set up some basic variables that we reference later in the code. 
+
+```python
+def __init__(self, provider: AMPProviderV2):
+    self.name = "Weather Distance"
+    self.provider = provider
+    self.set_output = False 
+
+    self.forecast_endpoint = "http://api.weatherapi.com/v1/forecast.json"
+    self.weather_key = "[Your WeatherAPI key here]"
+
+    self.distance_endpoint = "https://maps.googleapis.com/maps/api/distancematrix/json"
+    self.origin = "San Francisco"
+    self.units = "imperial"
+    self.distance_key = "[Your Google Maps key here]"
+    
+    self.provider.io.info(f"{self.name} tool started")
+```
+> :information_source: Protect your API keys! They are included here for ease of use and the assumption is that nobody will look at your version of the code. If you need to check your code int into version control, make sure to remove your keys!
+
+### Weather API Function
+To keep our code clean, we will write a function called `_get_weather` that takes a destination city, and calculates the distance from our origin which we set in the intialization step.
+
+We first use the `requests` library to make a GET request to the `WeatherAPI` REST endpoint, passing our destination, the number of days we're interested in (1), and our key as parameters. 
+
+With the results, we will use the `jsonpath_ext_rw` library to get the value we want out of the response with just a couple of lines of code.
+
+A very truncated example response might look like this:
+```json
+{
+    "location": {
+        "name": "London",
+        "region": "City of London, Greater London",
+        "country": "United Kingdom",
+    },
+    "current": {
+        "condition": {
+            "text": "Partly cloudy",
+            "icon": "//cdn.weatherapi.com/weather/64x64/day/116.png",
+            "code": 1003
+        },
+    },
+    "forecast": {
+        "forecastday": [
+            {
+                "date": "2023-05-02",
+                "date_epoch": 1682985600,
+                "day": {
+                    "mintemp_f": 52,
+                    "maxtemp_f": 75,
+                    "totalprecip_in": 0.33,
+                    "maxwind_mph": 15,
+                    "daily_will_it_rain": 1,
+                    "daily_chance_of_rain": 74,
+                },
+            }
+        ]
+    }
+}
+```
+You can see the entire response in [forecast.json](./assets/forecast.json).
+
+Notice how the values we want, `daily_chance_of_rain`, `maxwind_mph`, etc, are several levels deep in the JSON response. In Python can access this value with:
```

Comment:
TW review:

Notice how the values we want, `daily_chance_of_rain`, `maxwind_mph`, etc., are several levels deep in the JSON response. In Python, you can access this value with: